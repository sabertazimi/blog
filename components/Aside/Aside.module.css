.admonition.admonition {
  @apply mb-9 mt-0;
}

.admonition[role='alert'] div,
.admonition ol,
.admonition ul {
  @apply mb-3;
}

.admonition ol li::before {
  @apply mr-1;
}

.admonition div:last-child,
.admonition ol:last-child,
.admonition ul:last-child,
.admonition[role='alert'] li {
  @apply my-0;
}

.admonition[role='alert'] code {
  @apply text-sm;
}
