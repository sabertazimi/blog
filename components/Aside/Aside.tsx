import Alert from '@components/Alert'
import { cx } from '@components/utils'
import type { ReactNode } from 'react'
import styles from './Aside.module.css'

interface Props {
  type?: string
  title?: string
  children?: ReactNode
  className?: string
}

function normalizeType(type?: string) {
  switch (type) {
    case 'success':
    case 'tip':
      return 'success'
    case 'info':
    case 'note':
      return 'info'
    case 'warning':
    case 'caution':
      return 'warning'
    case 'error':
    case 'danger':
      return 'error'
    default:
      return 'info'
  }
}

function Aside({
  type,
  title,
  children,
  className,
  ...props
}: Props): JSX.Element {
  return (
    <Alert
      {...props}
      type={normalizeType(type)}
      message={title}
      description={children}
      showIcon
      className={cx(
        className,
        styles.admonition,
        'dark:bg-black dark:shadow-xl dark:shadow-primary',
      )}
    />
  )
}

export default Aside
