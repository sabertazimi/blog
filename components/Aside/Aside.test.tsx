import { render } from '@utils'
import Aside from './Aside'

describe('Aside', () => {
  const types = [
    'success',
    'tip',
    'info',
    'note',
    'warning',
    'caution',
    'error',
    'danger',
    'default',
  ]

  test('should render correctly (snapshot)', () => {
    const { container } = render(<Aside title="Aside">Aside</Aside>)

    expect(container).toMatchSnapshot()
  })

  test.each(types)(
    'should render different type correctly (snapshot)',
    type => {
      const { container } = render(
        <Aside type={type} title="Aside">
          Aside
        </Aside>
      )

      expect(container).toMatchSnapshot()
    }
  )
})
