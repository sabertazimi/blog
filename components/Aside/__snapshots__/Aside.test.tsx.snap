// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aside should render correctly (snapshot) 1`] = `
<div>
  <div
    class="ant-alert ant-alert-info ant-alert-with-description admonition dark:shadow-xl dark:shadow-primary"
    data-show="true"
    role="alert"
  >
    <span
      aria-label="info-circle"
      class="anticon anticon-info-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="info-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
        />
      </svg>
    </span>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        Aside
      </div>
      <div
        class="ant-alert-description"
      >
        Aside
      </div>
    </div>
  </div>
</div>
`;

exports[`Aside should render different type correctly (snapshot) 1`] = `
<div>
  <div
    class="ant-alert ant-alert-success ant-alert-with-description admonition dark:shadow-xl dark:shadow-primary"
    data-show="true"
    role="alert"
  >
    <span
      aria-label="check-circle"
      class="anticon anticon-check-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="check-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
        />
      </svg>
    </span>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        Aside
      </div>
      <div
        class="ant-alert-description"
      >
        Aside
      </div>
    </div>
  </div>
</div>
`;

exports[`Aside should render different type correctly (snapshot) 2`] = `
<div>
  <div
    class="ant-alert ant-alert-success ant-alert-with-description admonition dark:shadow-xl dark:shadow-primary"
    data-show="true"
    role="alert"
  >
    <span
      aria-label="check-circle"
      class="anticon anticon-check-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="check-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
        />
      </svg>
    </span>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        Aside
      </div>
      <div
        class="ant-alert-description"
      >
        Aside
      </div>
    </div>
  </div>
</div>
`;

exports[`Aside should render different type correctly (snapshot) 3`] = `
<div>
  <div
    class="ant-alert ant-alert-info ant-alert-with-description admonition dark:shadow-xl dark:shadow-primary"
    data-show="true"
    role="alert"
  >
    <span
      aria-label="info-circle"
      class="anticon anticon-info-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="info-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
        />
      </svg>
    </span>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        Aside
      </div>
      <div
        class="ant-alert-description"
      >
        Aside
      </div>
    </div>
  </div>
</div>
`;

exports[`Aside should render different type correctly (snapshot) 4`] = `
<div>
  <div
    class="ant-alert ant-alert-info ant-alert-with-description admonition dark:shadow-xl dark:shadow-primary"
    data-show="true"
    role="alert"
  >
    <span
      aria-label="info-circle"
      class="anticon anticon-info-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="info-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
        />
      </svg>
    </span>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        Aside
      </div>
      <div
        class="ant-alert-description"
      >
        Aside
      </div>
    </div>
  </div>
</div>
`;

exports[`Aside should render different type correctly (snapshot) 5`] = `
<div>
  <div
    class="ant-alert ant-alert-warning ant-alert-with-description admonition dark:shadow-xl dark:shadow-primary"
    data-show="true"
    role="alert"
  >
    <span
      aria-label="exclamation-circle"
      class="anticon anticon-exclamation-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="exclamation-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
        />
      </svg>
    </span>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        Aside
      </div>
      <div
        class="ant-alert-description"
      >
        Aside
      </div>
    </div>
  </div>
</div>
`;

exports[`Aside should render different type correctly (snapshot) 6`] = `
<div>
  <div
    class="ant-alert ant-alert-warning ant-alert-with-description admonition dark:shadow-xl dark:shadow-primary"
    data-show="true"
    role="alert"
  >
    <span
      aria-label="exclamation-circle"
      class="anticon anticon-exclamation-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="exclamation-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
        />
      </svg>
    </span>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        Aside
      </div>
      <div
        class="ant-alert-description"
      >
        Aside
      </div>
    </div>
  </div>
</div>
`;

exports[`Aside should render different type correctly (snapshot) 7`] = `
<div>
  <div
    class="ant-alert ant-alert-error ant-alert-with-description admonition dark:shadow-xl dark:shadow-primary"
    data-show="true"
    role="alert"
  >
    <span
      aria-label="close-circle"
      class="anticon anticon-close-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="close-circle"
        fill="currentColor"
        fill-rule="evenodd"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
        />
      </svg>
    </span>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        Aside
      </div>
      <div
        class="ant-alert-description"
      >
        Aside
      </div>
    </div>
  </div>
</div>
`;

exports[`Aside should render different type correctly (snapshot) 8`] = `
<div>
  <div
    class="ant-alert ant-alert-error ant-alert-with-description admonition dark:shadow-xl dark:shadow-primary"
    data-show="true"
    role="alert"
  >
    <span
      aria-label="close-circle"
      class="anticon anticon-close-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="close-circle"
        fill="currentColor"
        fill-rule="evenodd"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
        />
      </svg>
    </span>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        Aside
      </div>
      <div
        class="ant-alert-description"
      >
        Aside
      </div>
    </div>
  </div>
</div>
`;

exports[`Aside should render different type correctly (snapshot) 9`] = `
<div>
  <div
    class="ant-alert ant-alert-info ant-alert-with-description admonition dark:shadow-xl dark:shadow-primary"
    data-show="true"
    role="alert"
  >
    <span
      aria-label="info-circle"
      class="anticon anticon-info-circle ant-alert-icon"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="info-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
        />
      </svg>
    </span>
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        Aside
      </div>
      <div
        class="ant-alert-description"
      >
        Aside
      </div>
    </div>
  </div>
</div>
`;
